### PR TITLE
added param support

### PR DESCRIPTION
**Pull request by willmcgugan**
Adds query string params to FS URLs (work in progress).

### CODE REVIEW

**Comment by muffl0n** [comment]
Looks very promising to me! If you need any help just let me know! 
Thank you!

**Comment by muffl0n** [comment]
I would like to add this test to emphasize the usage of user and password for the target ftp server
```
    def test_parse_user_password_proxy(self):
        parsed = opener.parse('ftp://user:password@ftp.example.org?proxy=ftp.proxy.org')
        expected = opener.registry.ParseResult(
            'ftp',
            'user',
            'password',
            'ftp.example.org',
            {
                'proxy': 'ftp.proxy.org'
            },
            None
        )
        self.assertEqual(expected, parsed)
```

**Comment by willmcgugan** [comment]
@muffl0n @althonos This is in PyPi as `fs==1.0.10a2` if you would like to test.

Plan to do a full release tomorrow, if there are no issues.

File: fs/opener/parse.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
+"""
```
Comment (willmcgugan) [inline fs/opener/parse.py]: This is just factored out, no material changes.

**Pull request merged**